### PR TITLE
Route tests

### PR DESCRIPTION
**Pull request by mitjaziv**
- added unit tests to check routes functionalities (login/logout/auth)
- fixed request parameters parsing

@titpetric Please check if this commit `af1c811` is actual error or I am misunderstanding the usage of this.

### CODE REVIEW

**Comment by titpetric** [comment]
The commit in af1c811 seems valid, I forgot to copy in the filler on creation.

File: system/routes_test.go [hunk 1]
```diff
@@ -27,22 +44,187 @@ func TestUsers(t *testing.T) {
...
+			} `json:"response"`
+		}
+
+		var jr jsonResponse
+		err = json.NewDecoder(resp.Body).Decode(&jr)
+		assert(t, err == nil, "Error decoding response body: %+v", err)
+		assert(t, jr.Response.UserID != "0", "Expected userID not to be 0, got: %+v", jr.Response.UserID)
+		assert(t, jr.Response.Username == "johndoe", "Expected username 'johndoe', got: %+v", jr.Response.Username)
+	}
+
```
Comment (titpetric) [inline system/routes_test.go]: Test `/auth/check` here to see if login returns valid token.

File: system/routes_test.go [hunk 2]
```diff
@@ -27,22 +44,187 @@ func TestUsers(t *testing.T) {
...
+	// Send "Login" request with existing user.
+	{
+		form := url.Values{}
+		form.Add("username", "johndoe")
+		form.Add("password", "johndoe123")
+
+		req, err := http.NewRequest("POST", "http://localhost/auth/login", strings.NewReader(form.Encode()))
+		req.Header.Add("Content-Type", "application/x-www-form-urlencoded")
```
Comment (titpetric) [inline system/routes_test.go]: I'd like you to test with a json post body here separately (`application/json` content-type).

File: system/routes_test.go (hunk 1 above)
Comment (mitjaziv) [reply system/routes_test.go]: Changed.

**Pull request closed without merging**